### ISSUE TITLE
[java] SwitchStmtsShouldHaveDefault false positive on enum inside enums

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#2946]
**Affects PMD Version:** 6.29.0

**Rule:** [SwitchStmtsShouldHaveDefault](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#switchstmtsshouldhavedefault)

**Description:**

If a switch-case over a enum is used inside the enum itself, PMD wrongly requests to add a default case, even though the switch statement is exhaustive. The switch is over `this`:

**Code Sample demonstrating the issue:**

```java
package myPackage;

public enum GradeSystem {

    SysA, SysB;

    public boolean isValid() {
      switch (this) { // false-positive
        case SysA:
        case SysB:
          return true;
      }
      
      return false;
    }
}
```

Result:
`myPackage/GradeSystem.java:8:	Switch statements should have a default label`

**Expected outcome:**

*   Does PMD report a violation, where there shouldn't be one? -> false-positive

**Running PMD through:** CLI

**Comment by adangel** [comment]
This has been fixed with PMD 7.0.0-rc1.
